### PR TITLE
Add new entry on file structure

### LINKED ISSUE #779: [Entry] General: File structure
### Reason/inspiration (optional)

Write a General concept entry about file/directory structures, what they, and how they are used by computers to organize information.

Recommendation: Find ways to hyperlink to the existing [file format](https://www.codecademy.com/resources/docs/general/file-formats) and [file path](https://www.codecademy.com/resources/docs/general/file-paths) entries.

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [x] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by YanisaHS**
### Description

I added the file-system-structure concept and file-system-structure.md file inside. I also added file-system-structure.png to the media folder. The file:

1. Introduces the file system structure
2. Elaborates on the root, directories, and files
3. Gives an example of a file system structure using the Filesystem Hierarchy Standard (FHS) in Linux

Just noting that I also changed "file structure" to "file system structure" because I thought it was more descriptive. I can change it back though if the old name is preferred!

Closes #779 

### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Review by SSwiniarski — APPROVED** [review]
@YanisaHS, I did a first review and it looks good to go right to review 2.

File: content/general/concepts/file-system-structure/file-system-structure.md
Comment (yangc95) [inline content/general/concepts/file-system-structure/file-system-structure.md:22]:
```suggestion
The root or root directory is the highest level directory of the file system structure. In macOS, Linux, and other UNIX-like operating systems, the root is represented with a forward slash (`/`). In Windows, the root is represented by the drive name and a backslash (`\`).
```

File: content/general/concepts/file-system-structure/file-system-structure.md
Comment (yangc95) [inline content/general/concepts/file-system-structure/file-system-structure.md:30]:
```suggestion
Files are storage units on a computer which contain information on file contents, instructions needed to read them, and where they are stored. A file’s location in the directory system is referred to as the [file path](https://www.codecademy.com/resources/docs/general/file-paths), and the [file format](https://www.codecademy.com/resources/docs/general/file-formats) or type determines how data in the file is created, stored, and read.
```

File: content/general/concepts/file-system-structure/file-system-structure.md
Comment (yangc95) [inline content/general/concepts/file-system-structure/file-system-structure.md:34]:
```suggestion
The Filesystem Hierarchy Standard (FHS) is one example of a defined file system structure. The FHS is for [UNIX](https://www.codecademy.com/resources/docs/general/unix) systems, and is most popularly used in [Linux](https://www.codecademy.com/resources/docs/general/linux). It is maintained by the Linux Foundation, and most Linux distributions follow the FHS.
```

File: content/general/concepts/file-system-structure/file-system-structure.md
Comment (yangc95) [inline content/general/concepts/file-system-structure/file-system-structure.md:40]:
```suggestion
- `/bin` stores all binary executable files.
- `/tmp` stores all temporary files.
- `/sys` stores information on the Linux kernel and various system devices.
```
```suggestion
- `/bin`: Stores all binary executable files
- `/tmp`: Stores all temporary files
- `/sys`: Stores information on the Linux kernel and various system devices
```

File: content/general/concepts/file-system-structure/file-system-structure.md
Comment (yangc95) [inline content/general/concepts/file-system-structure/file-system-structure.md:42]:
```suggestion
The purpose of using a defined file system structure like the FHS is so users and software can predict the locations of files and directories. This makes it easier to compile, install, and run software.
```

**Review by yangc95 — COMMENTED** [review]
@YanisaHS I went ahead and did a second review!

Please let me know if you have any concerns with any of my comments and tag me once you've looked them over.

**Comment by YanisaHS** [comment]
Thank you for the feedback @yangc95 ! I've finished reviewing the changes for this one now 🥳

**Review by yangc95 — APPROVED** [review]

File: content/general/concepts/file-system-structure/file-system-structure.md
Comment (Dusch4593) [inline content/general/concepts/file-system-structure/file-system-structure.md:16]:
```suggestion
The **file system structure** defines how information in key (existing and future) directories and files are organized and how stored in an operating system.

Although different operating systems may have different standards, all major operating systems have similar characteristics and use a hierarchical structure of directories branching off the root directory. The image below illustrates an example of the hierarchy in a sample file system structure.
```

File: content/general/concepts/file-system-structure/file-system-structure.md
Comment (Dusch4593) [inline content/general/concepts/file-system-structure/file-system-structure.md:22]:
```suggestion
The root directory, or "root", is the highest-level directory of the file system structure. In macOS, [Linux](https://www.codecademy.com/resources/docs/open-source/linux), and other [UNIX](https://www.codecademy.com/resources/docs/general/unix)-like operating systems, the root is represented with a forward slash (`/`). In Windows, the root is represented by the drive name and a backslash (`\`).
```

File: content/general/concepts/file-system-structure/file-system-structure.md
Comment (Dusch4593) [inline content/general/concepts/file-system-structure/file-system-structure.md:34]:
```suggestion
The Filesystem Hierarchy Standard (FHS) is one example of a defined file system structure. The FHS is for UNIX systems, and is most popularly used in Linux. It is maintained by the Linux Foundation, and most Linux distributions follow the FHS.
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
Some comments/suggestions 😄

**Comment by YanisaHS** [comment]
Thank you for the feedback @Dusch4593 ! Done! 🥳

**Comment by yangc95** [comment]
@Dusch4593 did we have anything else to change here before merging?

**Comment by Dusch4593** [comment]
@yangc95 @YanisaHS This seems ready to merge. 🚀 

I made one little change in 91e878d46d6768700ad38d07fdd9f99334ec0b74 where I broke up the paragraph on line # 30 into separate ones (2-3 sentences each). The last two sentences seemed to explore a new idea in subdirectories and how they relate to their parents and to one another.

**Review by Dusch4593 — APPROVED** [review]

**Comment by Dusch4593** [comment]
Thank you for another awesome entry, @YanisaHS! 🚀 🎊

The live entry can be viewed here: https://www.codecademy.com/resources/docs/general/file-system-structure

**Pull request merged**
